#! /usr/bin/env python

PACKAGE='youbot_manipulation_vision'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *


gen = ParameterGenerator()
#       Name                    					Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("vision_threshold_binary",   				int_t,	  0, 							    "The binary threshold value used to seperate black objects from the white background.",             95,    	0, 	  255)
gen.add("vision_canny_threshold",    				int_t, 	  0,    							"The canny threshold value used to distinguish the smooth cylinder from the rough one.",            40,     0,    255)
gen.add("vision_corner_threshold",					int_t, 	  0,								"The number of corners before to detect a small cylinder as rough small cylinder.",					25,		0,	  300)
gen.add("vision_keep_object_count",					int_t, 	  0,    							"The maximum number of objects too keep in the detected objects", 								     4,     0,    50)
gen.add("vision_min_objekt_size",     				int_t, 	  0,    							"The minimum size of an object before the object gets filtered out",                    			 4000,  0,    100000)
exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "ManipulationVision"))